src/main/java/edu/ie3/datamodel/io/connectors/InfluxDbConnector.java

```diff
@@ -31,43 +31,72 @@
-  private static final String INFLUXDB_DATABASE_NAME = "ie3_in";
-  private final String databaseName;
   private final String scenarioName;
-  private final String url;
+  private final InfluxDB session;
 
   /**
    * Initializes a new InfluxDbConnector with the given url, databaseName and scenario name.
    *
    * @param url the connection url for the influxDB database
-   * @param databaseName the name of the database to which the connection should be established
    * @param scenarioName the name of the simulation scenario which will be used in influxDB
    *     measurement names
+   * @param databaseName the name of the database the session should be set to
+   * @param createDb true if the connector should create the database if it doesn't exist yet, false
+   *     otherwise
+   * @param logLevel log level of the {@link InfluxDB.LogLevel} logger
+   * @param batchOptions write options to write batch operations
    */
-  public InfluxDbConnector(String url, String databaseName, String scenarioName) {
-    this.url = url;
-    this.databaseName = databaseName;
+  public InfluxDbConnector(
+      String url,
+      String databaseName,
+      String scenarioName,
+      boolean createDb,
+      InfluxDB.LogLevel logLevel,
+      BatchOptions batchOptions) {
+    this(InfluxDBFactory.connect(url), scenarioName, databaseName, createDb);
+    this.session.setLogLevel(logLevel);
+    this.session.enableBatch(batchOptions);
+  }
+
+  /**
+   * Initializes a new InfluxDbConnector with the given influxDb session, the databaseName and
+   * scenario name.
+   *
+   * @param session the influxdb session that should be managed by this connector
+   * @param scenarioName the name of the scenario
+   * @param databaseName the name of the database the session should be set to
+   * @param createDb true if the connector should create the database if it doesn't exist yet, false
+   *     otherwise
+   */
+  public InfluxDbConnector(
+      InfluxDB session, String scenarioName, String databaseName, boolean createDb) {
     this.scenarioName = scenarioName;
+    this.session = session;
+
+    session.setDatabase(databaseName);
+    if (createDb) createDb(databaseName);
   }
 
   /**
-   * Initializes a new InfluxDbConnector with the given url and databaseName and no scenario name.
-   * Consider using a scenario name if you plan to persist results using this connector.
+   * Initializes a new InfluxDbConnector with the given url, databaseName and scenario name.
    *
    * @param url the connection url for the influxDB database
    * @param databaseName the name of the database to which the connection should be established
+   * @param scenarioName the name of the simulation scenario which will be used in influxDB
+   *     measurement names
    */
-  public InfluxDbConnector(String url, String databaseName) {
-    this(url, databaseName, null);
+  public InfluxDbConnector(String url, String databaseName, String scenarioName) {
```

Comment:
The scenario name was meant to be used for output data, to reference the used scenario. If we use only weather, we don't need the scenario name. If we require it, would it make sense to offer a default constant? 
Either an empty scenario `public static final String NO_SCENARIO = "no_scenario";` or a weather specific one `public static final String WEATHER_SCENARIO = "weather_scenario";`.